core/main/src/service/user_grants.rs

```diff
@@ -121,6 +121,102 @@ impl GrantState {
         None
     }
 
+    pub async fn sync_grant_map_with_grant_policy(&self, platform_state: &PlatformState) {
+        let grant_entries_to_remove = Self::fetch_grant_entry_to_remove(platform_state);
+        for (app_id, entry_set) in grant_entries_to_remove.iter() {
+            for entry in entry_set {
+                Self::force_delete_user_grant_from_all_sources(
+                    platform_state,
+                    app_id.clone(),
+                    entry,
+                )
+                .await;
+            }
+        }
+    }
+
+    fn fetch_grant_entry_to_remove(
+        platform_state: &PlatformState,
+    ) -> HashMap<String, HashSet<GrantEntry>> {
+        let mut grant_entries_to_remove: HashMap<String, HashSet<GrantEntry>> = HashMap::new();
+        let grant_policies_map = if let Some(grant_policies) =
+            platform_state.get_device_manifest().get_grant_policies()
+        {
+            grant_policies
+        } else {
+            HashMap::default()
+        };
+        let grant_entries = platform_state
+            .cap_state
+            .grant_state
+            .grant_app_map
+            .read()
+            .unwrap();
+        for (app_id, app_entries) in grant_entries.value.iter() {
+            let mut grant_entry_set_to_remove: HashSet<GrantEntry> = HashSet::new();
+            for grant_entry in app_entries.iter() {
+                if !grant_policies_map.contains_key(&grant_entry.capability) {
+                    grant_entry_set_to_remove.insert(grant_entry.clone());
+                }
+            }
+            if !grant_entry_set_to_remove.is_empty() {
+                grant_entries_to_remove.insert(app_id.clone(), grant_entry_set_to_remove);
+            }
+        }
+        debug!(
+            "sync: grant entries to be removed {:#?}",
+            grant_entries_to_remove
+        );
+        grant_entries_to_remove
+    }
+
+    async fn force_delete_user_grant_from_all_sources(
+        platform_state: &PlatformState,
+        app_id: String,
+        entry: &GrantEntry,
+    ) {
+        // Delete app grant in local storage and grant state by app id
+        let grant_entry_to_remove =
+            Self::force_delete_user_grant(platform_state, app_id.clone(), entry);
+
+        if let Some(gc_opt) = grant_entry_to_remove {
+            // Remove grant entry from cloud
+            let mut gc = gc_opt;
+            gc.status = None;
+
+            GrantPolicyEnforcer::send_usergrants_for_cloud_storage(
```

Comment:
Do we need to send for all cases or only for when storage type is set to 'sync' ?